azurelinuxagent/common/utils/restutil.py

```diff
@@ -354,7 +354,14 @@ def http_request(method,
                  max_retry=None,
                  retry_codes=None,
                  retry_delay=DELAY_IN_SECONDS,
-                 redact_data=False):
+                 redact_data=False,
+                 return_raw_response=False):
```

Comment:
The error handling done by http_request mixes logic for several APIs, and makes hard (or impossible) to handle specific errors for new APIs (in this case vmSettings). Our mock infrastructure is based on this method, though, so I decided to add a new parameter to bypass all the error handling here.

Eventually this method needs to be rewritten/replaced to do better error handling.